geopandas/io/file.py

```diff
@@ -626,7 +626,7 @@ def infer_schema(df):
     from collections import OrderedDict
 
     # TODO: test pandas string type and boolean type once released
-    types = {"Int64": "int", "string": "str", "boolean": "bool"}
+    types = {"Int32": "int32", "Int64": "int", "string": "str", "boolean": "bool"}
```

Comment:
Note that only `pd.Int32Dtype()` columns will have the schema `int32`, while regular `np.int32` columns would still become `int`. Would this be ok? cc @seisman.

Reply:
I'm not sure. https://github.com/geopandas/geopandas/pull/2464 looks more promising.

Reply:
#2464 is using this:
```suggestion
    types = {
        "int64": "int",
        "int32": "int32",
        "int16": "int32",
        "int8": "int32",
        "uint64": "int",
        "uint32": "int",
        "uint16": "int",
        "uint8": "int",
        "string": "str",
        "boolean": "bool",
    }
```
So either casting to int or int32. I'm not sure how this would affect other OGR drivers though.

Reply:
Ok, I've changed things in 6859d6c1ebc2410d6e44141eb0b04ff1476c8a49 to only cast `np.int32` and `pd.Int32Dtype()`:

```python
types = {
    "Int32": "int32",  # nullable pd.Int32Dtype()
    "int32": "int32",  # non-nullable np.int32
    "Int64": "int",
    "string": "str",
    "boolean": "bool",
}
```

As mentioned at https://github.com/geopandas/geopandas/pull/2950#issuecomment-1620814235, we'll handle the casting of int64 to int32 in PyGMT at https://github.com/GenericMappingTools/pygmt/pull/2592, so that this doesn't break other OGR drivers that may support saving int64 columns.